clang/lib/AST/ItaniumCXXABI.cpp

```diff
@@ -236,8 +236,7 @@ class ItaniumCXXABI : public CXXABI {
 
   CallingConv getDefaultMethodCallConv(bool isVariadic) const override {
     const llvm::Triple &T = Context.getTargetInfo().getTriple();
-    if (!isVariadic && T.isWindowsGNUEnvironment() &&
-        T.getArch() == llvm::Triple::x86)
+    if (!isVariadic && T.isOSCygMing() && T.getArch() == llvm::Triple::x86)
```

Comment:
I know this PR is closed, but for the record: this particular change appears to be wrong: on i686 cygwin with this change calls to libstdc++ basic_string::_M_assign crashed, but with this change backed out it's much happier.